### ISSUE TITLE
Gradual updates with Pinot helm chart

### ISSUE THREAD

**Issue by ivanvy-wix** [apache/pinot#9514]
The [recommended order for updating Pinot’s components](https://docs.pinot.apache.org/operators/operating-pinot/upgrading-pinot-cluster) is hard to follow with the current helm chart, because there’s only one global version and no way to update a single thing at the time.

The simplest thing would be to have (empty by default) image overrides that one could set during upgrades:
```yaml
# values.yaml
...
controller:
  image:
      repository:
      tag:
...
```
```yaml
# /templates/controller/statefulset.yaml
...
      containers:
      - name: controller
        securityContext:
          {{- toYaml .Values.controller.securityContext | nindent 10 }}
        image: "{{ .Values.controller.image.repository | default .Values.image.repository }}:{{ .Values.controller.image.tag | default .Values.image.tag }}"
...
```

Is it worth doing? Does the gradual components upgrade worth this additional complexity?

**Comment by Jackie-Jiang** [comment]
cc @xiangfu0
